lib/commands/reshim.sh

```diff
@@ -93,29 +120,106 @@ generate_shims_for_version() {
     local install_type="version"
     local version="${version_info[0]}"
   fi
+  space_separated_list_of_bin_paths="$(list_plugin_bin_paths "$plugin_name" "$version" "$install_type")"
+  IFS=' ' read -a all_bin_paths <<< "$space_separated_list_of_bin_paths"
 
-  local install_path=$(get_install_path $plugin_name $install_type $version)
+  local install_path=$(get_install_path "$plugin_name" "$install_type" "$version")
 
-  if [ -f ${plugin_path}/bin/list-bin-paths ]; then
-    local space_separated_list_of_bin_paths=$(
-      export ASDF_INSTALL_TYPE=$install_type
-      export ASDF_INSTALL_VERSION=$version
-      export ASDF_INSTALL_PATH=$install_path
-      bash ${plugin_path}/bin/list-bin-paths
-    )
+  for bin_path in "${all_bin_paths[@]}"; do
+    for executable_file in $install_path/$bin_path/*; do
+      # because just $executable_file gives absolute path; We don't want version hardcoded in shim
+      local executable_path_relative_to_install_path=$bin_path/$(basename "$executable_file")
+      if [ -x "$executable_file" ]; then
+          write_shim_script "$plugin_name" "$version" "$executable_path_relative_to_install_path"
+      fi
+    done
+  done
+}
+
+shim_still_exists() {
+  local shim_name=$1
+  local install_path=$2
+  local space_separated_list_of_bin_paths=$3
+  IFS=' ' read -a all_bin_paths <<< "$space_separated_list_of_bin_paths"
+
+
+  for bin_path in "${all_bin_paths[@]}"; do
+    if [ -x "$install_path/$bin_path/$shim_name" ]; then
+      return 0
+    fi
+  done
+  return 1
+}
```

Comment:
I might be missing something here, but would `plugin_bin_still_exists` be a better name? This is checking for the presence of a binary in the plugin installation directory right?